lib/helpers.js

```diff
@@ -42,20 +42,48 @@ const format = (event, options = { ignoreSelection: false }) => {
   const selectedText = editor.getSelectedText();
   const isTransformingFile = options.ignoreSelection || !selectedText;
 
-  const textToTransform = isTransformingFile ? editor.getText() : selectedText;
+  const currentScope = getCurrentScope();
+  const isEmbedded = ['text.html.vue', 'text.html.basic'].includes(currentScope);
+  if (isEmbedded) {
+    const re = /<script\b[^>]*>([\s\S]*?)(?=<\/script>)/gi
+    editor.scan(re, iter => {
+      // only transform first match
+      iter.stop();
```

Comment:
Figured it out. Have to use `editor.backwardsScanInBufferRange` rather than `editor.scan` since transforming the text in the forward direction causes misalignment with the buffer range. Now I can get multiple script blocks working.